notebooks/GWAS_blood_pressure_p3/main.py

```diff
@@ -0,0 +1,649 @@
+               'pheno.blood_pressure_test_bp_diastolic': 'bp_diastolic',
+               'pheno.blood_pressure_test_bp_systolic': 'bp_systolic',
+               }
+    samples_traits_for_analysis.rename(columns=col_map, inplace=True)
+
+with herzog.Cell("python"):
+    # Check that this worked
+    # Note that because we used the nwd_id as the key to match the phenotypic to genotypic data,
+    # nwd_id is now the firt column
+    samples_traits_for_analysis.head()
+
+with herzog.Cell("markdown"):
+    """
+    ## Write the phenotype data to a new file
+    Write the phenotype data exported from the outputs of Hail to a new file for use in the GENESIS workflows. Hail may have removed individuals if they did not have genotype data associated with them.
+    """
+
+with herzog.Cell("python"):
+    samples_traits_for_analysis.to_csv(phenotype_out, index=False)
+
+with herzog.Cell("markdown"):
+    """
+    ## Move the phenotype and GRM files to the workspace bucket
+    """
+
+with herzog.Cell("python"):
+    #!gsutil cp {phenotype_out} {bucket + phenotype_out}
+    #!gsutil cp {kinship_out} {bucket + kinship_out}
+    pass
+
+with herzog.Cell("markdown"):
+    """
+    ## Generate a new Terra data model called "sample_set" and add all of our derived data files to this entity.
+
+    The last step in the analysis is to update **your** workspace data table with the data generated in this notebook. For downstream analysis, you'll want to set up all the needed parameters and input data directly in the data table. As you will see, analysis workflows look to the data table for configuration and inputs.
+
+    Recall that each *sample* in the data table corresponded to a single individual. To run a GWAS requires a set of samples, or in Terra terms, a **sample_set**. This entity should include everything you might need to run a workflow for genotype-phenotype modeling: a list of samples to include, phenotype data, a list of covariates, an outcome, a genetic relatedness matrix, and a label. The particular workflow that you will soon run looks for inputs in specific columns of the data table.
+
+    Below is a helpful function for generating a sample set in the correct format so you will have success in the next part this template. To look at the full code, click the arrow at the top left of the cell.
+
+    In a GWAS, you may iterate through an analysis multiple times by filtering your data differently or including/excluding covariates. For each separate run, you can cange the values in the columns, specifically the value in the sample_set_id column to reflect a new analysis. You will see that you can save each notebook you used for each iteration as well.
+    """
+
+with herzog.Cell("python"):
+    # Make a new "sample set" entity type and create an entity within called "systloicbp"
+
+    #column headers
+    cols = ['entity:sample_set_id',
+            'phenotypes',
+            'phenotypes_notebook',
```

Comment:
Do you want to have 2 columns for saving each notebook? One for phenotypes_notebook and one for hail_notebook? Also, I think if you don't actually load the phenotype.ipynb from your bucket back into memory this won't consistently work.